internal/backends/qt/qt_accessible.rs

```diff
@@ -0,0 +1,439 @@
+                let mut descendents = i_slint_core::accessibility::accessible_descendents(
+                        &*(root_item as *mut ItemRc));
+                descendents.shrink_to_fit();
+                Box::into_raw(Box::new(descendents)) as _
+            });
+        }
+
+        size_t count() {
+            return rust!(Descendents_count [rustDescendents: *mut core::ffi::c_void as "void*"] -> usize as "size_t" {
+               let vector = Box::from_raw(rustDescendents as *mut Vec<ItemRc>);
+               let result = vector.len();
+               std::mem::forget(vector);
+
+               result
+            });
+        }
+
+        void* itemAt(size_t index) {
+            return rust!(Descendents_itemAt [rustDescendents: *mut core::ffi::c_void as "void*",
+                                             index: usize as "size_t"]
+                    -> *mut core::ffi::c_void as "void*" {
+                let mut vector = Box::from_raw(rustDescendents as *mut Vec<ItemRc>);
+                let item_rc = vector[index].clone();
+                let mut item_weak = Box::new(item_rc.downgrade());
+
+                let result = core::ptr::addr_of_mut!(*item_weak);
+
+                std::mem::forget(vector);
+                std::mem::forget(item_weak);
+
+                result as _
+            });
+        }
+
+        QAccessible::Role roleAt(size_t index) {
+            return rust!(Descendents_roleAt [rustDescendents: *mut core::ffi::c_void as "void*",
+                                             index: usize as "size_t"]
+                    -> u32 as "QAccessible::Role" {
+                let vector = Box::from_raw(rustDescendents as *mut Vec<ItemRc>);
+                let result = match vector[index].accessible_role() {
+                    i_slint_core::items::AccessibleRole::none => 0x00, // QAccessible::NoRole
+                    i_slint_core::items::AccessibleRole::text => 0x29, // QAccessible::StaticText
+                    i_slint_core::items::AccessibleRole::button => 0x2b, // QAccessible::Button
+                    i_slint_core::items::AccessibleRole::checkbox => 0x2c, // QAccessible::CheckBox
+                };
+                std::mem::forget(vector);
+
+                result
+            });
+        }
+
+        ~Descendents() {
+            rust!(Descendents_dtor [rustDescendents: *mut core::ffi::c_void as "void*"] {
+               Box::from_raw(rustDescendents as *mut Vec<ItemRc>);
+            });
+        }
+
+    private:
+        void *rustDescendents;
+    };
+
+    void *root_item_for_window(void *rustWindow) {
+        return rust!(root_item_for_window_ [rustWindow: *const core::ffi::c_void as "void*"]
+                -> *mut core::ffi::c_void as "void*" {
+            let window = &*(rustWindow as *const i_slint_core::window::Window);
+
+            let root_item = Box::new(ItemRc::new(window.component(), 0).downgrade());
+            Box::into_raw(root_item) as _
+        });
```

Comment:
You can use the rust! to cast directly

```suggestion
        return rust!(root_item_for_window_ [rustWindow: &i_slint_core::window::Window as "void*"]
                -> *mut ItemRc as "void*" {
            let root_item = Box::new(ItemRc::new(rustWindow.component(), 0).downgrade());
            Box::into_raw(root_item)
        });
```

And in other places too

Reply:
I only discovered that half-way through writing this. Looks like I forgot to update this.